pkg/test/framework/components/stackdriver/kube.go

```diff
@@ -0,0 +1,150 @@
+	stackdriverPort      = 8091
+)
+
+var (
+	_ Instance  = &kubeComponent{}
+	_ io.Closer = &kubeComponent{}
+)
+
+type kubeComponent struct {
+	id        resource.ID
+	env       *kube.Environment
+	ns        namespace.Instance
+	forwarder testKube.PortForwarder
+}
+
+func newKube(ctx resource.Context) (Instance, error) {
+	env := ctx.Environment().(*kube.Environment)
+	c := &kubeComponent{
+		env: env,
+	}
+	c.id = ctx.TrackResource(c)
+	var err error
+	scopes.CI.Info("=== BEGIN: Deploy Stackdriver ===")
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("stackdriver deployment failed: %v", err) // nolint:golint
+			scopes.CI.Infof("=== FAILED: Deploy Stackdriver ===")
+			_ = c.Close()
+		} else {
+			scopes.CI.Info("=== SUCCEEDED: Deploy Stackdriver ===")
+		}
+	}()
+
+	c.ns, err = namespace.New(ctx, namespace.Config{
+		Prefix: stackdriverNamespace,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("could not create %s Namespace for Stackdriver install; err:%v", stackdriverNamespace, err)
+	}
+
+	// apply stackdriver YAML
+	yamlContent, err := ioutil.ReadFile(environ.StackdriverInstallFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read %s, err: %v", environ.StackdriverInstallFilePath, err)
+	}
+
+	if err := env.ApplyContents(c.ns.Name(), string(yamlContent)); err != nil {
+		return nil, fmt.Errorf("failed to apply rendered %s, err: %v", environ.StackdriverInstallFilePath, err)
+	}
+
+	fetchFn := env.Accessor.NewSinglePodFetch(c.ns.Name(), "app=stackdriver")
+	pods, err := env.Accessor.WaitUntilPodsAreReady(fetchFn)
+	if err != nil {
+		return nil, err
+	}
+	pod := pods[0]
+
+	forwarder, err := env.Accessor.NewPortForwarder(pod, 0, stackdriverPort)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := forwarder.Start(); err != nil {
+		return nil, err
+	}
+	c.forwarder = forwarder
+	scopes.Framework.Debugf("initialized stackdriver port forwarder: %v", forwarder.Address())
+
+	return c, nil
+}
+
+func (c *kubeComponent) ListTimeSeries() ([]*monitoringpb.TimeSeries, error) {
+	client := http.Client{
+		Timeout: 5 * time.Second,
+	}
+	resp, err := client.Get("http://" + c.forwarder.Address() + "/timeseries")
+	if err != nil {
+		return []*monitoringpb.TimeSeries{}, err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return []*monitoringpb.TimeSeries{}, err
+	}
+	var r monitoringpb.ListTimeSeriesResponse
+	err = jsonpb.UnmarshalString(string(body), &r)
+	if err != nil {
+		return []*monitoringpb.TimeSeries{}, err
+	}
+	var ret []*monitoringpb.TimeSeries
+	for _, t := range r.TimeSeries {
+		// Remove fields that do not need verification
+		t.Points = nil
+		t.Resource = nil
+		ret = append(ret, t)
+	}
+	return ret, nil
+}
+
+func (c *kubeComponent) ID() resource.ID {
+	return c.id
+}
+
+// Close implements io.Closer.
+func (c *kubeComponent) Close() error {
+	scopes.CI.Infof("Deleting Stackdriver Install")
+	_ = c.env.DeleteNamespace(stackdriverNamespace)
```

Comment:
Can stackdriver deployment be structured like Bookinfo app deployment? You would not have to track deletion of namespace in that case?

Also, looks like deployment is failing in test because of following error:
standard_init_linux.go:211: exec user process caused "exec format error"

Reply:
Yeah I tried it out and indeed there is no need to explicitly delete the namespace.